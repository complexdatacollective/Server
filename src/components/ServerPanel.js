import React, { Component } from 'react';
import { PanelItem } from '../components';
import Server from '../utils/Server';

const defaultServerOverview = {
  ip: 'x.x.x.x',
  clients: 0,
  uptime: 0,
  publicKey: '',
};

class ServerPanel extends Component {
  constructor() {
    super();

    this.state = {
      version: '0.0.0',
      notes: '',
    };

    this.server = new Server();

    this.server.on('SERVER_STATUS', (data) => {
      this.setState({ serverOverview: data });
    });

    this.server.requestServerStatus();
  }

  render() {
    const { serverOverview } = this.state;

    const overview = { ...defaultServerOverview, ...serverOverview };
    return (
      <div className="server-panel">
        <PanelItem label="IP" value={JSON.stringify(overview.ip)} />
        <PanelItem label="Clients" value={overview.clients} />
        <PanelItem label="Uptime" value={overview.uptime} />
        <PanelItem label="Public Key" value={overview.publicKey} />
      </div>
    );
  }
}

export default ServerPanel;
