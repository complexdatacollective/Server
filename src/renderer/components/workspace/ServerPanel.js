import React from 'react';
import PropTypes from 'prop-types';
import DeviceStatus from '../../containers/DeviceStatus';
import NetworkStatus from '../../containers/NetworkStatus';

const ServerPanel = ({ className }) => (
  <div className={`server-panel ${className}`}>
    <div className="server-panel__wrapper">
      <NetworkStatus />
      <DeviceStatus />
    </div>
  </div>
);

ServerPanel.defaultProps = {
  className: '',
};

ServerPanel.propTypes = {
  className: PropTypes.string,
};

export {
  ServerPanel as UnwrappedServerPanel,
};

export default ServerPanel;
