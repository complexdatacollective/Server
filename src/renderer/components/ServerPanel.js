import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { PanelItem } from '../components';
import withApiClient from './withApiClient';

class ServerPanel extends Component {
  constructor() {
    super();
    this.state = {
      version: '0.0.0',
    };
  }

  componentWillMount() {
    this.getServerHealth();
  }

  componentDidUpdate() {
    this.getServerHealth();
  }

  getServerHealth() {
    const { apiClient } = this.props;
    if (!apiClient || this.state.serverOverview) {
      return;
    }
    apiClient.get('/health')
      .then((resp) => {
        const {
          deviceApiPort,
          hostname,
          ip,
          isAdvertising,
          mdnsIsSupported,
          uptime,
        } = resp.serverStatus;
        let mdnsStatus = isAdvertising ? 'Active' : 'Pending';
        if (!mdnsIsSupported) {
          mdnsStatus = 'Unsupported';
        }
        this.setState({
          serverOverview: {
            hostname,
            ip: ip && ip.address,
            deviceApiPort,
            mdnsStatus,
            uptime,
          },
        });
      })
      .catch(() => {
        this.setState({
          serverOverview: {},
        });
      });
  }

  render() {
    const { serverOverview } = this.state;
    const { className } = this.props;
    const overview = { ...serverOverview };
    const uptimeDisplay = overview.uptime && `${parseInt(overview.uptime / 1000 / 60, 10)}m`;
    return (
      <div className={`server-panel ${className}`}>
        <PanelItem label="Local Server IP" value={overview.ip || 'Offline'} />
        <PanelItem label="Server Port" value={overview.deviceApiPort || '-'} />
        <PanelItem label="Uptime" value={uptimeDisplay || '-'} />
        <PanelItem label="Server Hostname" value={overview.hostname || '-'} />
        <PanelItem label="Service Advertising" value={overview.mdnsStatus || '-'} />
      </div>
    );
  }
}

ServerPanel.defaultProps = {
  apiClient: null,
  className: '',
};

ServerPanel.propTypes = {
  apiClient: PropTypes.object,
  className: PropTypes.string,
};

export default withApiClient(ServerPanel);

export {
  ServerPanel as UnwrappedServerPanel,
};
