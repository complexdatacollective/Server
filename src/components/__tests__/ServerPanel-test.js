/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { WebSocket } from 'mock-socket';
import ServerPanel from '../ServerPanel';

console.log('hi');
console.log(WebSocket);
global.WebSocket = WebSocket;

const mockServerOverview = {
  ip: 'x.x.x.x',
  clients: 0,
  uptime: 0,
  publicKey: '',
};

describe('<ServerPanel />', () => {
  it('should render', () => {
    const subject = shallow((
      <ServerPanel serverOverview={mockServerOverview} />
    ));

    expect(subject).toMatchSnapshot();
  });
});
