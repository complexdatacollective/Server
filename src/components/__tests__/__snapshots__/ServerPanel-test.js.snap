// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServerPanel /> should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ServerPanel
    serverOverview={
        Object {
            "clients": 0,
            "ip": "x.x.x.x",
            "publicKey": "",
            "uptime": 0,
          }
    }
/>,
  "nodes": Array [
    <ServerPanel
      serverOverview={
            Object {
                  "clients": 0,
                  "ip": "x.x.x.x",
                  "publicKey": "",
                  "uptime": 0,
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <IPCComponent
        serverOverview={
                Object {
                        "clients": 0,
                        "ip": "x.x.x.x",
                        "publicKey": "",
                        "uptime": 0,
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": IPCComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "serverOverview": Object {
            "clients": 0,
            "ip": "x.x.x.x",
            "publicKey": "",
            "uptime": 0,
          },
        },
        "refs": Object {},
        "state": Object {
          "serverOverview": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ServerPanel
          serverOverview={
                    Object {
                              "clients": 0,
                              "ip": "x.x.x.x",
                              "publicKey": "",
                              "uptime": 0,
                            }
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <ServerPanel
          serverOverview={
                    Object {
                              "clients": 0,
                              "ip": "x.x.x.x",
                              "publicKey": "",
                              "uptime": 0,
                            }
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <IPCComponent
    serverOverview={
        Object {
            "clients": 0,
            "ip": "x.x.x.x",
            "publicKey": "",
            "uptime": 0,
          }
    }
/>,
}
`;
